src/LitEmoji.php

```diff
@@ -197,8 +197,16 @@ public static function config(string $property, $value): void
      */
     public static function removeEmoji(string $source): string
     {
-        $content = self::encodeShortcode($source);
+        /*$content = self::encodeShortcode($source);
         $content = preg_replace('/\:\w+\:/', '', $content);
+        return $content;*/
+        return self::replaceEmoji($source, '');
+    }
+    
+    public static function replaceEmoji(string $source, string $replacement):string
+    {
+        $content = self::encodeShortcode($source);
+        $content = preg_replace('/\:\w+\:/', $replacement, $content);
```

Comment:
I feel that only known shortcodes should be replaced; if the content already contains such patterns for other purposes then they should remain unaffected.

This should be simple with [preg_replace_callback](https://www.php.net/manual/en/function.preg-replace-callback.php).

Reply:
You mean:
```
$content = preg_replace_callback($pattern, static function($matches) use ($replacement){
            return $replacement;
        }, $content);
```
?

Reply:
Yes, however the return would look something like

`return in_array($parsedShortcode, $supportedShortcodes);`

The idea being that if it's just a word surrounded by `:` and not the name of an actual emoji, it remains unchanged.